agent/engine/docker_image_manager.go

```
@@ -175,6 +174,10 @@ func (imageManager *dockerImageManager) addContainerReferenceToExistingImageStat
 	imageState, ok := imageManager.getImageState(container.ImageID)
 	if ok {
 		imageState.UpdateImageState(container)
+		err := imageManager.dataClient.SaveImageState(imageState)
+		if err != nil {
+			return !ok
```

since the error is not returned, i think we should log it here for visibility